test/functional/notifiers_controller_test.rb

```diff
@@ -0,0 +1,93 @@
+require "test_helper"
+
+class NotifiersControllerTest < ActionController::TestCase
+  context "when not logged in" do
+    setup do
+      @user = create(:user)
+      get :show
+    end
+    should redirect_to("the sign in page") { sign_in_path }
+  end
+
+  context "when logged in" do
+    setup do
+      @user = create(:user)
+      sign_in_as(@user)
+    end
+
+    context "when user owns a gem with more than MFA_REQUIRED_THRESHOLD downloads" do
+      setup do
+        @rubygem = create(:rubygem)
+        @ownership = create(:ownership, rubygem: @rubygem, user: @user)
+        GemDownload.increment(
+          Rubygem::MFA_REQUIRED_THRESHOLD + 1,
+          rubygem_id: @rubygem.id
+        )
+        @request.cookies[:mfa_required] = "true"
+      end
+
+      redirect_scenarios = {
+        "GET to show" => [:show, { method: "GET" }],
+        "PATCH to update" => [:update, { method: "PATCH" }],
+        "PUT to update" => [:update, { method: "PUT" }]
```

Comment:
nit: to mimic a realistic request, the params should be added to these requests

Reply:
I can't figure this out. @ownership is nil if i put it in redirect_scenarios, even if I move the @ownership setup up one context. If I put redirect_scenarios as @redirect_scenarios in the setup, it becomes undefined instead. Do you want to give it a try?

Reply:
`redirect_scenarios` is outside of a setup or should block, so it can't access any instance variables. I think using 1 for the id is good enough 👍